pkg/server/test/property_value_aggregate_test.go

```diff
@@ -17,19 +20,40 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/grafana/iot-sitewise-datasource/pkg/sitewise/client/mocks"
-	"github.com/grafana/iot-sitewise-datasource/pkg/testdata"
 )
 
-func Test_propertyValueAggregateHappyCase(t *testing.T) {
-	propAggs := testdata.GetIoTSitewisePropAggregateVals(t, testDataRelativePath("property-aggregate-values.json"))
-	propDesc := testdata.GetIotSitewiseAssetProp(t, testDataRelativePath("describe-asset-property-raw-wind.json"))
+func Test_property_value_aggregate_query_by_asset_id_and_property_id(t *testing.T) {
 	mockSw := &mocks.SitewiseClient{}
-	mockSw.On("BatchGetAssetPropertyAggregatesPageAggregation", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(&propAggs, nil)
-	mockSw.On("DescribeAssetPropertyWithContext", mock.Anything, mock.Anything).Return(&propDesc, nil)
+	mockSw.On(
+		"BatchGetAssetPropertyAggregatesPageAggregation",
+		mock.Anything,
+		mock.MatchedBy(func(input *iotsitewise.BatchGetAssetPropertyAggregatesInput) bool {
+			entries := *input.Entries[0]
+			return *entries.EntryId == "1assetid-aaaa-2222-bbbb-3333cccc4444" &&
+				*entries.AssetId == "1assetid-aaaa-2222-bbbb-3333cccc4444" &&
+				*entries.PropertyId == "11propid-aaaa-2222-bbbb-3333cccc4444" &&
+				*entries.AggregateTypes[0] == "SUM"
+		}),
+		mock.Anything,
+		mock.Anything,
```

Comment:
do you think it would be easier to read if we named some of these inputs? I'm not sure if this is stylistically weird to do with go but I guess it feels like it'd be easier to follow this test if we were like:
```
mockSw.On(
		"BatchGetAssetPropertyAggregatesPageAggregation",
		mockRequestContext,
		mockInput,
		mockPages,
		mockMaxResults,
	)
```

wdyt? Maybe I'm being silly/nitpicky?

Reply:
These are just really huge tests since they're at the highest level (feels a bit end to end); I cannot yet find the ideal balance between something "digestable" and something where the reader doesn't get re-directed all the time to find the underlying values. 

It's my opinion that by not extracting the inputs into variables, we make it easier to connect the query input to what's being passed to some of the methods, and so on down the chain.

I think this test remains difficult to read, but mostly because is testing a very long chain of external API calls. Some ideas: maybe duplicate this test to only make assertions on one method at a time (so we can just use mock.Anything) or we could separate the tests and test at lower levels.

I _think_ this test a reasonable improvement over the previous input which was read from very far away files and sometimes quite lengthy, and an output which was also written to some file far away. I wouldn't be surprised to see further improvements iteratively as we dive into this repo more. 

Anyway, I will ultimately support whatever Kevin thinks is best!

Reply:
i think i'll leave it as is for now and we can iterate on further improvements later.

i think keeping it as `mock.Anything` has the advantage of being more clear since it's what's provided by the mocking library and also when I hover over it I see the following so i can see a particular argument isn't important for the given test case.

```
const mock.Anything untyped string = "mock.Anything"
Anything is used in Diff and Assert when the argument being tested shouldn't be taken into consideration.
```

also a more minor reason, if we ever modify the arguments to these functions, we'd need to check every single place to make sure name still makes sense.